build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,369 @@
+  endif()
+  add_dependencies(iree-test-deps "${_NAME}")
+endfunction()
+
+# iree_benchmark_suite_module_test()
+#
+# Creates a test using iree-run-module to run a benchmark suite module.
+#
+# The function supports cross compile and checks the benchmark suite prefixes
+# based on the cross compile platform settings, so there is no matching bzl rule
+# until there is a bazel --build_config for cross compile targets in place.
+#
+# Parameters:
+#   NAME: Name of the target
+#   BENCHMARK_MODULE_SRC: IREE module from benchmark_suite.
+#   DRIVER: Driver to run the module with. If specified, it has to be consistent
+#       with the one provided by the benchmark_suite's flagfile.
+#   RUNNER_ARGS: additional args to pass to iree-run-module. The driver
+#       and input file are passed automatically.
+#   EXPECTED_OUTPUT: A file of expected output to compare with the output from
+#       iree-run-module
+#   LABELS: Additional labels to apply to the test. The package path and
+#       "driver=${DRIVER}" are added automatically.
+#   XFAIL: When set for x86_64|android-arm64-v8a|riscv64-Linux|riscv32-Linux,
+#       the test is expected to fail for the particular platforms/architectures
+#       for various reasons, e.g., upstream llvm backend.
+#   UNSUPPORTED_PLATFORMS: Platforms
+#       (android-arm64-v8a|riscv64-Linux|riscv32-Linux) not supported by the
+#       test target. The target will be skipped during crosscompile.
+#   TIMEOUT: (optional) Test timeout.
+#
+# Examples:
+#
+# iree_benchmark_suite_module_test(
+#   NAME
+#     person_detect_int8_correctness_test
+#   BENCHMARK_MODULE_SRC
+#     "TFLite/PersonDetect-int8"
+#   DRIVER
+#     "local-sync"
+#   RUNNER_ARGS
+#     "--entry_function=main"
+#     "--function_input=1x96x96x1xi8=0"
+#   EXPECTED_OUTPUT
+#     "1x2xi8=[72 -72]"
+#   UNSUPPORTED_PLATFORMS
+#     "android-arm64-v8a"
+# )
+#
+# iree_benchmark_suite_module_test(
+#   NAME
+#     mobilenet_v1_fp32_correctness_test
+#   BENCHMARK_MODULE_SRC
+#     "TFLite/MobileNetV1-fp32,imagenet"
+#   DRIVER
+#     "local-sync"
+#   RUNNER_ARGS
+#     "--entry_function=main"
+#     "--function_input=1x224x224x3xf32=0"
+#   EXPECTED_OUTPUT
+#     "mobilenet_v1_fp32_expected_output.txt"
+#   UNSUPPORTED_PLATFORMS
+#     "android-arm64-v8a"
+#     "riscv32-Linux"
+# )
+function(iree_benchmark_suite_module_test)
+if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME;BENCHMARK_MODULE_SRC;DRIVER;EXPECTED_OUTPUT;TIMEOUT"
+    "RUNNER_ARGS;LABELS;XFAIL;UNSUPPORTED_PLATFORMS"
+    ${ARGN}
+  )
+
+  if(CMAKE_CROSSCOMPILING AND "hostonly" IN_LIST _RULE_LABELS)
+    return()
+  endif()
+
+  iree_get_platform(_PLATFORM)
+  if(_PLATFORM IN_LIST _RULE_UNSUPPORTED_PLATFORMS)
+    return()
+  endif()
+
+  # Benchmark suite needs to be stored at the location of
+  # `IREE_BENCHMARK_SUITE_DIR` or the test target is bypassed.
+  if(_RULE_BENCHMARK_MODULE_SRC AND NOT DEFINED IREE_BENCHMARK_SUITE_DIR)
```

Comment:
Isn't `_RULE_BENCHMARK_MODULE_SRC` mandatory?

Reply:
Update the condition to check `IREE_BENCHMARK_SUITE_DIR` only.